# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1489583618 14400
#      Wed Mar 15 09:13:38 2017 -0400
# Node ID df0fb2cb820be2efecc03e47aaeae4554b79516e
# Parent  ea143f4f76a51d8e8147fba64c6f0eabb5da7cac
add release dates to NEWS files

* NEWS, etc/NEWS.1, etc/NEWS.2, etc/NEWS.3: Add known release dates to
  Summary lines.

* etc/CHECKLIST: Note that release date info should be added to NEWS
  file Summary lines.  Add item for updating release history wiki
  page.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -1,10 +1,10 @@
-Summary of important user-visible changes for version 4.4:
----------------------------------------------------------
+Summary of important user-visible changes for version 4.4 (yyyy-mm-dd):
+----------------------------------------------------------------------
 
  ** On systems with 64-bit pointers, --enable-64 is now the default and
     Octave always uses 64-bit indexing.  However, if the configure
     script determines that the BLAS library uses 32-bit integers, then
     operations using the following libraries are limited to arrays with
     dimensions that are smaller than 2^31 elements:
 
       BLAS  LAPACK  QRUPDATE  SuiteSparse  ARPACK
@@ -83,18 +83,18 @@ Summary of important user-visible change
     that it defined (DECLARE_OCTAVE_ALLOCATOR, DEFINE_OCTAVE_ALLOCATOR,
     and DEFINE_OCTAVE_ALLOCATOR2).
 
  ** The rectangle and ellipse annotation property "edgecolor" has been
     deprecated and will be removed from Octave 4.8 (or whatever version
     is the second major release after 4.4).  Use the property "color"
     instead.
 
-Summary of bugs fixed for version 4.2.1:
----------------------------------------
+Summary of bugs fixed for version 4.2.1 (2017-02-22):
+----------------------------------------------------
 
 Using the bug numbers listed below, find bug reports on the web using
 the URL https://savannah.gnu.org/bugs/?NNNNN
 
  ** guarantee returning std::string from tilde_expand functions (bug #50234)
 
  ** workaround segfault in file_stat (bug #50234)
 
@@ -207,18 +207,18 @@ Documentation bugs fixed:
  ** small fixes to docstrings (bug #49733)
 
  ** change text describing demo plots to reflect new ColorOrder (bug #49288)
 
 Other bugs fixed:
 
  ** add missing classdef test files (bug #49819)
 
-Summary of important user-visible changes for version 4.2:
----------------------------------------------------------
+Summary of important user-visible changes for version 4.2 (2016-11-13):
+----------------------------------------------------------------------
 
  ** The parser has been extended to accept, but ignore, underscore
     characters in numbers.  This facilitates writing more legible code
     by using '_' as a thousands separator or to group nibbles into bytes
     in hex constants.
 
     Examples: 1_000_000 == 1e6  or  0xDE_AD_BE_EF
 
@@ -453,18 +453,18 @@ Summary of important user-visible change
     gripe_XXX to either err_XXX if error is called or warn_XXX if
     warning is called.  The gripe_XXX functions are deprecated and will
     be removed in version 4.6.
 
  ** New configure option, --enable-address-sanitizer-flags, to build
     Octave with memory allocator checks (similar to those in valgrind)
     built in.
 
-Summary of important user-visible changes for version 4.0:
----------------------------------------------------------
+Summary of important user-visible changes for version 4.0 (2015-05-23):
+----------------------------------------------------------------------
 
  ** A graphical user interface is now the default when running Octave
     interactively.  The start-up option --no-gui will run the familiar
     command line interface, and still allows use of the GUI dialogs and
     qt plotting toolkit.  The option --no-gui-libs runs a minimalist
     command line interface that does not link with the Qt libraries and
     uses the fltk toolkit for plotting.
 
diff --git a/etc/CHECKLIST b/etc/CHECKLIST
--- a/etc/CHECKLIST
+++ b/etc/CHECKLIST
@@ -1,15 +1,17 @@
 Things to do before making a release:
 
   * Update the version number and release date in configure.ac.
 
   * Update version number and release year in CITATION.
 
-  * Update the NEWS file.
+  * Update the NEWS file.  Include release date in Summary header.
 
   * Update the NEWS-<VERSION>.html, index.in, news.in, and download.in
     files for the web site.
 
   * Copy tarballs to ftp.gnu.org.
 
   * Post a release announcement to the help@octave.org and
     info-gnu@gnu.org lists.
+
+  * Update http://wiki.octave.org/Release_History page.
diff --git a/etc/NEWS.1 b/etc/NEWS.1
--- a/etc/NEWS.1
+++ b/etc/NEWS.1
@@ -1,10 +1,10 @@
-Summary of changes for version 1.1.1:
-------------------------------------
+Summary of changes for version 1.1.1 (1995-02-23):
+-------------------------------------------------
 
   * New built-in variables, default_return_value and
     define_all_return_values.
 
     If define_all_return_values is set to "false", Octave does not do
     anything special for return values that are left undefined, and
     you will get an error message if you try to use them.  For
     example, if the function
@@ -46,18 +46,18 @@ Summary of changes for version 1.1.1:
   * The break statement may also be used to exit a function, for
     compatibility with Matlab.
 
   * The directory tree for installing Octave is now closer to
     conforming with the current GNU standards.
 
   * More bug fixes.
 
-Summary of changes for version 1.1.0:
-------------------------------------
+Summary of changes for version 1.1.0 (1995-01-12):
+-------------------------------------------------
 
   * Octave now requires g++ 2.6.3 or later.  This change is necessary
     to make template instantiations cleaner, and to avoid having to
     have special cases in the code for earlier versions of gcc.
 
   * A new data structure type has been added.  The implementation uses
     an associative array with indices limited to strings, but the
     syntax is more like C-style structures.  here are some examples of
@@ -579,36 +579,36 @@ Summary of changes for version 1.1.0:
   * New option --verbose.  If Octave is invoked with --verbose and not
     --silent, a message is printed if an octaverc file is read while
     Octave is starting.
 
   * An improved configure script generated by Autoconf 2.0.
 
   * Lots of bug fixes.
 
-Summary of changes for version 1.0:
-----------------------------------
+Summary of changes for version 1.0 (1994-02-17):
+-----------------------------------------------
 
   * C-style I/O functions now handle files referenced by name or by
     number more consistently.
 
-Summary of changes for version 0.83:
------------------------------------
+Summary of changes for version 0.83 (1994-02-13):
+------------------------------------------------
 
   * Loading global symbols should work now.
 
   * Clearing the screen doesn't reprint the prompt unnecessarily.
 
   * The operations <complex scalar> OP <real matrix> for OP == +, -,
     *, or ./ no longer crash Octave.
 
   * More portability and configuration fixes.
 
-Summary of changes for version 0.82:
------------------------------------
+Summary of changes for version 0.82 (1994-02-08):
+------------------------------------------------
 
   * Octave now comes with a reference card.
 
   * The manual has been improved, but more work remains to be done.
 
   * The atanh function now works for complex arguments.
 
   * The asin, acos, acosh, and atanh functions now work properly when
@@ -630,18 +630,18 @@ Summary of changes for version 0.82:
   * Octave no longer crashes if you type ^D at the beginning of a line
     in the middle of defining a loop or if statement.
 
   * On AIX systems, Back off on indexing DiagArray via Proxy class to
     avoid gcc (or possibly AIX assembler?) bug.
 
   * Various other bug and portability fixes.
 
-Summary of changes for version 0.81:
------------------------------------
+Summary of changes for version 0.81 (1994-01-28):
+------------------------------------------------
 
   * Octave no longer dumps core if you try to define a function in
     your .octaverc file.
 
   * Fixed bug in Array class that resulted in bogus off-diagonal
     elements when computing eigenvalue and singular value
     decompositions.
 
@@ -649,18 +649,18 @@ Summary of changes for version 0.81:
     at least with some versions of Sun's f77.  This bug was introduced
     in 0.80, when I changed LSODE to allow the user to abort the
     integration from within the RHS function.
 
   * Fixed bug that prevented global attribute of variables from being
     saved with save(), and another that prevented load() from working
     at all.
 
-Summary of changes for version 0.80:
------------------------------------
+Summary of changes for version 0.80 (1994-01-18):
+------------------------------------------------
 
   * I have started working on a manual for the C++ classes.  At this
     point, it is little more than a list of function names.  If you
     would like to volunteer to help work on this, please contact
     maintainers@octave.org.
 
   * The patterns accepted by the save and clear commands now work like
     file name globbing patterns instead of regular expressions.  I
@@ -787,32 +787,32 @@ Summary of changes for version 0.80:
     will now blink the cursor to show matching parens.
 
   * By default, include files are now installed in
     $prefix/include/octave instead of $prefix/include.
 
   * Octave now uses a config.h file instead of putting all defines on
     the compiler command line.
 
-Summary of changes for version 0.79:
------------------------------------
+Summary of changes for version 0.79 (1993-11-08):
+------------------------------------------------
 
   * New control systems functions:
 
      dgram -- Returns the discrete controllability and observability gramian.
      dlqr  -- Discrete linear quadratic regulator design.
      dlqe  -- Discrete linear quadratic estimator (Kalman Filter) design.
      c2d   -- Convert continuous system description to discrete time
               description assuming zero-order hold and given sample time.
 
   * The max (min) functions can now return the index of the max (min)
     value as a second return value.
 
-Summary of changes for version 0.78:
------------------------------------
+Summary of changes for version 0.78 (1993-11-05):
+------------------------------------------------
 
   * Octave's handling of global variables has been completely
     rewritten.  To access global variables inside a function, you must
     now declare them to be global within the function body.  Likewise,
     if you do not declare a variable as global at the command line,
     you will not have access to it within a function, even if it is
     declared global there.  For example, given the function
 
@@ -925,27 +925,27 @@ Summary of changes for version 0.78:
     like filename globbing.  For example, given the variables `a',
     `aa', and `a1', the command `save a*' saves `a' and `aa' but not
     `a1'.  To match all variables beginning with `a', you must use an
     expression like `a.*' (match all sequences beginning with `a'
     followed by zero or more characters).
 
   * Line and column information is included in more error messages.
 
-Summary of changes for version 0.77:
------------------------------------
+Summary of changes for version 0.77 (1993-10-23):
+------------------------------------------------
 
   * Improved help.  The command `help -i topic' now uses the GNU Info
     browser to display help for the given topic directly from the
     Texinfo documentation.
 
   * New function: chol -- Cholesky factorization.
 
-Summary of changes for version 0.76:
------------------------------------
+Summary of changes for version 0.76 (1993-10-05):
+------------------------------------------------
 
   * Better run-time error messages.  Many now include line and column
     information indicating where the error occurred.  Octave will also
     print a traceback for errors occurring inside functions. If you
     find error messages that could use improvement, or errors that
     Octave fails to catch, please send a bug report to
     bug@octave.org.
 
@@ -955,18 +955,18 @@ Summary of changes for version 0.76:
 
       gplot "sin (x)" w l, data w p
 
     to work (assuming that data is a variable containing a matrix of
     values).
 
   * Long options (--help, --version, etc.) are supported.
 
-Summary of changes for version 0.75:
------------------------------------
+Summary of changes for version 0.75 (1993-09-15):
+------------------------------------------------
 
   * The documentation is much more complete, but still could use a lot
     of work.
 
   * The history function now prints line numbers by default.  The
     command `history -q' will  omit them.
 
   * The clear function now accepts regular expressions.
@@ -1001,18 +1001,18 @@ Summary of changes for version 0.75:
       are             -- Solve algebraic Ricatti equation.
       dare            -- Solve discrete-time algebraic Ricatti equation.
       lqe             -- Kalman filter design for continuous linear system.
       lqr             -- Linear Quadratic Regulator design.
       lyap            -- Solve Lyapunov equation.
       dlyap           -- Solve discrete Lyapunov equation.
       tzero           -- Compute the transmission zeros of [A,B,C,D].
 
-Summary of changes for version 0.74:
------------------------------------
+Summary of changes for version 0.74 (1993-07-20):
+------------------------------------------------
 
   * Formal parameters to functions are now always considered to be
     local variables, so things like
 
       global x = 0
       global y = 0
       function y = f (x) x = 1; y = x; end
       f (x)
@@ -1022,18 +1022,18 @@ Summary of changes for version 0.74:
 
   * Multiple assignment expressions are now allowed to take indices,
     so things like
 
       octave:13> [a([1,2],[3,4]), b([5,6],[7,8])] = lu ([1,2;3,4])
 
     will work correctly.
 
-Summary of changes for version 0.73:
------------------------------------
+Summary of changes for version 0.73 (1993-07-10):
+------------------------------------------------
 
   * Saving and loading global variables works correctly now.
 
   * The save command no longer saves built-in variables.
 
   * Global variables are more reliable.
 
   * Matrices may now have one or both dimensions zero, so that
@@ -1074,18 +1074,18 @@ Summary of changes for version 0.73:
   * The bodies of functions, and the for, while, and if commands are
     now allowed to be empty.
 
   * Support for Gill and Murray's QPSOL has been added.  Like NPSOL,
     QPSOL is not freely redistributable either, so you must obtain
     your own copy to be able to use this feature.  More information
     about where to find QPSOL and NPSOL are in the file README.NLP.
 
-Summary of changes for version 0.72:
------------------------------------
+Summary of changes for version 0.72 (1993-06-10):
+------------------------------------------------
 
   * For numeric output, columns are now lined up on the decimal point.
     (This requires libg++-2.3.1 or later to work correctly).
 
   * If octave is running interactively and the output intended for the
     screen is longer than one page and a pager is available, it is
     sent to the pager through a pipe.  You may specify the program to
     use as the pager by setting the variable PAGER.  PAGER may also
@@ -1216,18 +1216,18 @@ Summary of changes for version 0.72:
       #!/bin/sh
       OCTAVE_HOME=/foo/bar/baz
       export OCTAVE_HOME
       exec octave.bin
 
     to be invoked as octave.
 
 
-Summary of changes for version 0.71:
------------------------------------
+Summary of changes for version 0.71 (1993-04-15):
+------------------------------------------------
 
   * Much improved plotting facility.  With this release, Octave does
     not require a specially modified version of gnuplot, so gnuplot
     sources are no longer distributed with Octave.  For a more
     detailed description of the new plotting features, see the file
     PLOTTING.
 
   * New plotting commands:
@@ -1289,18 +1289,18 @@ Summary of changes for version 0.71:
     input like [1,2;3,4,[5,6]] now produces a diagnostic message.
 
   * The cd, save, and load commands now do tilde expansion.
 
   * It's now possible to clear global variables and functions by name.
 
   * Use of clear inside functions is now a parse error.
 
-Summary of changes for version 0.70:
------------------------------------
+Summary of changes for version 0.70 (1993-03-08):
+------------------------------------------------
 
   * Better parse error diagnostics.  For interactive input, you get
     messages like
 
       octave:1> a = 3 + * 4;
 
       parse error:
 
@@ -1325,18 +1325,18 @@ Summary of changes for version 0.70:
 
   * Better support for missing math functions.
 
   * User preferences are now cached in a global struct so we don't
     have to do a symbol table lookup each time we need to know what
     they are.  This should mean slightly improved performance for
     evaluating expressions.
 
-Summary of changes for version 0.69:
------------------------------------
+Summary of changes for version 0.69 (1993-02-23):
+------------------------------------------------
 
   * Multiple assignments are now possible, so statements like
 
       a = b = c = 3;
       a = b = c = [1,2;3,4];
 
     or
 
@@ -1355,26 +1355,26 @@ Summary of changes for version 0.69:
     commands to execute.
 
   * Better detection and conditional compilation of IEEE functions
     isinf, finite, and isnan.
 
   * Replacements for acosh, asinh, atanh, and gamma from the BSD math
     library for those systems that don't have them.
 
-Summary of changes for version 0.68:
------------------------------------
+Summary of changes for version 0.68 (1993-02-16):
+------------------------------------------------
 
   * New functions:
 
       eval  -- evaluate a string as a sequence of Octave commands.
       input -- print a prompt and get user input.
 
-Summary of changes for version 0.67:
------------------------------------
+Summary of changes for version 0.67 (1993-02-09):
+------------------------------------------------
 
   * New functions:
 
       find -- return the indices of nonzero elements.
 
   * Zero-one style indexing now works.  For example,
 
       a = [1,2,3,4];
@@ -1415,18 +1415,18 @@ Summary of changes for version 0.67:
       if ~[], 1, else 0, end
 
   * Octave no longer converts input like `3.2 i' or `3 I' to complex
     constants directly because that causes problems inside square
     brackets, where spaces are important.  This abbreviated notation
     *does* work if there isn't a space between the number and the i,
     I, j, or J.
 
-Summary of changes for version 0.66:
------------------------------------
+Summary of changes for version 0.66 (1993-01-28):
+------------------------------------------------
 
   * Logical unary not operator (~ or !) now works for complex.
 
   * Left division works.
 
   * Right and left element by element division should work correctly
     now.
 
@@ -1437,27 +1437,27 @@ Summary of changes for version 0.66:
   * The min and max functions should work correctly for two arguments.
 
   * Improved (I hope!) configuration checks.
 
   * Octave is now installed as octave-M.N, where M and N are version
     numbers, and octave is a link to that file.  This makes it
     possible to have more than one version of the interpreter installed.
 
-Summary of changes for version 0.63:
------------------------------------
+Summary of changes for version 0.63 (1993-01-14):
+------------------------------------------------
 
   * The reshape function works again.
 
   * Octave now converts input like `3.2i' or `3 I' or `2.3e5 j' to be
     complex constants directly, rather than requiring an expression
     like `3.3 * i' to be evaluated.
 
-Summary of changes for version 0.61:
------------------------------------
+Summary of changes for version 0.61 (1993-01-10):
+------------------------------------------------
 
   * Octave has been successfully compiled using gcc 2.3.3 and libg++ 2.3.
     on a 486 system running Linux.
 
   * The win_texas_lotto function is now called texas_lotto (it's a
     script file, and win_texas_lotto.m is too long for some Linux and
     System V systems).
 
diff --git a/etc/NEWS.2 b/etc/NEWS.2
--- a/etc/NEWS.2
+++ b/etc/NEWS.2
@@ -1,10 +1,10 @@
-Summary of changes for version 2.1.x:
-------------------------------------
+Summary of changes for version 2.1.x (1997-06-05 -- 2006-03-20):
+---------------------------------------------------------------
 
   * Given a matrix, X, and a boolean index, idx, of the same shape as
     X, X(idx) and X(idx) = RHS now work no matter what the value of
     do_fortran_indexing is.
 
   * If you are using GNU Emacs 19.34 or earlier, you will need to add
     the following code to your ~/.emacs file in order to use Emacs
     Octave mode:
@@ -497,23 +497,23 @@ Summary of changes for version 2.1.x:
     value is 256.
 
   * Octave now uses kpathsea 3.2.
 
   * New configure option, --enable-readline.
 
   * New configure option, --enable-static.
 
-Summary of changes for version 2.0.7:
-------------------------------------
+Summary of changes for version 2.0.7 (1997-06-04):
+-------------------------------------------------
 
   This is a bug-fixing release.  There are no new user-visible features.
 
-Summary of changes for version 2.0.6:
-------------------------------------
+Summary of changes for version 2.0.6 (1997-05-27):
+-------------------------------------------------
 
   This is primarily a bug-fixing release.  There are only a few new
   user-visible features.
 
   * The new built-in variable default_eval_print_flag controls whether
     Octave prints the results of commands executed by eval() that do
     not end with semicolons.  The default is 1.
 
@@ -527,41 +527,41 @@ Summary of changes for version 2.0.6:
     number of microseconds to delay in the parent process after
     forking.  By default on gnu-win32 systems, it's set to 500000 (1/2
     second).  On other systems, the default value is 0.  Delaying for
     a short time in the parent after forking seems to avoid problems
     in which communicating with subprocesses via pipes would sometimes
     cause Octave to hang.  I doubt that the delay is really the right
     solution.  If anyone has a better idea, I'd love to hear it.
 
-Summary of changes for version 2.0.5:
-------------------------------------
+Summary of changes for version 2.0.5 (1997-03-01):
+-------------------------------------------------
 
   * A `switch' statement is now available.  See the Statements chapter
     in the manual for details.
 
   * Commands like ls, save, and cd may now also be used as formal
     parameters for functions.
 
   * More tests.
 
-Summary of changes for version 2.0.4:
-------------------------------------
+Summary of changes for version 2.0.4 (1997-02-20):
+-------------------------------------------------
 
   * It is now possible to use commands like ls, save, and cd as simple
     variable names.  They still cannot be used as formal parameters
     for functions, or as the names of structure variables.  Failed
     assignments leave them undefined (you can recover the original
     function definition using clear).
 
   * Is is now possible to invoke commands like ls, save, and cd as
     normal functions (for example, load ("foo", "x", "y", "z")).
 
-Summary of changes for version 2.0.3:
-------------------------------------
+Summary of changes for version 2.0.3 (1997-02-18):
+-------------------------------------------------
 
   * The manual has been completely revised and now corresponds much
     more closely to the features of the current version.
 
   * The return value for assignment expressions is now the RHS since
     that is more consistent with the way other programming languages
     work.  However, Octave still prints the entire LHS value so that
 
@@ -578,36 +578,36 @@ Summary of changes for version 2.0.3:
 
       z = x(2) = 1
 
     sets z to 1 (not [ 0, 1 ] as in previous versions of Octave).
 
   * It is now much easier to make binary distributions.  See the
     Binary Distributions section of the manual for more details.
 
-Summary of changes for version 2.0.2:
-------------------------------------
+Summary of changes for version 2.0.2 (1997-01-27):
+-------------------------------------------------
 
   * Octave now stops executing commands from a script file if an error
     is encountered.
 
   * The return, and break commands now cause Octave to quit executing
     commands from script files.  When used in invalid contexts, the
     break, continue, and return commands are now simply ignored
     instead of producing parse errors.
 
   * size ("") is now [0, 0].
 
   * New functions:
 
       sleep   -- pause execution for a specified number of seconds
       usleep  -- pause execution for a specified number of microseconds
 
-Summary of changes for version 2.0:
-----------------------------------
+Summary of changes for version 2.0 (1996-12-10):
+-----------------------------------------------
 
   * The set and show commands for setting and displaying gnuplot
     parameters have been replaced by gset and gshow.  This change will
     probably break lots of things, but it is necessary to allow for
     compatibility with the Matlab graphics and GUI commands in a
     future version of Octave.  (For now, the old set and show commands
     do work, but they print an annoying warning message to try to get
     people to switch to using gset.)
diff --git a/etc/NEWS.3 b/etc/NEWS.3
--- a/etc/NEWS.3
+++ b/etc/NEWS.3
@@ -1,10 +1,10 @@
-Summary of important user-visible changes for version 3.8:
----------------------------------------------------------
+Summary of important user-visible changes for version 3.8 (2013-12-27):
+----------------------------------------------------------------------
 
   ** One of the biggest new features for Octave 3.8 is a graphical user
      interface.  It is the one thing that users have requested most
      often over the last few years and now it is almost ready.  But
      because it is not quite as polished as we would like, we have
      decided to wait until the 4.0.x release series before making the
      GUI the default interface (until then, you can use the --force-gui
      option to start the GUI).
@@ -392,18 +392,18 @@ Summary of important user-visible change
       CC_VERSION  (now GCC_VERSION)
       CXX_VERSION (now GXX_VERSION)
 
     The internal class <Octave_map> has been deprecated in Octave 3.8 and will
     be removed from Octave 3.12 (or whatever version is the second major
     release after 3.8).  Replacement classes are <octave_map> (struct array)
     or <octave_scalar_map> for a single structure.
 
-Summary of important user-visible changes for version 3.6:
----------------------------------------------------------
+Summary of important user-visible changes for version 3.6 (2012-01-15):
+----------------------------------------------------------------------
 
  ** The PCRE library is now required to build Octave.  If a pre-compiled
     package does not exist for your system, you can find PCRE sources
     at http://www.pcre.org
 
  ** The ARPACK library is no longer distributed with Octave.
     If you need the eigs or svds functions you must provide an
     external ARPACK through a package manager or by compiling it
@@ -608,29 +608,29 @@ Summary of important user-visible change
       corrcoef           studentize
       __error_text__     sylvester_matrix
       error_text
 
  ** The following functions have been modified for Matlab compatibility:
 
       randperm
 
-Summary of important user-visible changes for version 3.4.3:
------------------------------------------------------------
+Summary of important user-visible changes for version 3.4.3 (2011-10-10):
+------------------------------------------------------------------------
 
  ** Octave 3.4.3 is a bug fixing release.
 
-Summary of important user-visible changes for version 3.4.2:
------------------------------------------------------------
+Summary of important user-visible changes for version 3.4.2 (2011-06-24):
+------------------------------------------------------------------------
 
  ** Octave 3.4.2 fixes some minor installation problems that affected
     version 3.4.1.
 
-Summary of important user-visible changes for version 3.4.1:
------------------------------------------------------------
+Summary of important user-visible changes for version 3.4.1 (2011-06-15):
+------------------------------------------------------------------------
 
  ** Octave 3.4.1 is primarily a bug fixing release.
 
  ** IMPORTANT note about binary incompatibility in this release:
 
     Binary compatibility for all 3.4.x releases was originally planned,
     but this is impossible for the 3.4.1 release due to a bug in the way
     shared libraries were built in Octave 3.4.0.  Because of this bug,
@@ -670,18 +670,18 @@ Summary of important user-visible change
     functions are now private.
 
  ** ChangeLog files in the Octave sources are no longer maintained
     by hand.  Instead, there is a single ChangeLog file generated from
     the Mercurial version control commit messages.  Older ChangeLog
     information can be found in the etc/OLD-ChangeLogs directory in the
     source distribution.
 
-Summary of important user-visible changes for version 3.4:
----------------------------------------------------------
+Summary of important user-visible changes for version 3.4 (2011-02-08):
+----------------------------------------------------------------------
 
  ** BLAS and LAPACK libraries are now required to build Octave.  The
     subset of the reference BLAS and LAPACK libraries has been removed
     from the Octave sources.
 
  ** The ARPACK library is now distributed with Octave so it no longer
     needs to be available as an external dependency when building
     Octave.
@@ -1107,18 +1107,18 @@ Summary of important user-visible change
     The following functions have been deprecated in Octave 3.4 and will
     be removed from Octave 3.8 (or whatever version is the second major
     release after 3.4):
 
       autocor  cellidx   gammai     is_global  replot     values
       autocov  dispatch  glpkmex    krylovb    saveimage
       betai    fstat     intwarning perror     strerror
 
-Summary of important user-visible changes for version 3.2:
----------------------------------------------------------
+Summary of important user-visible changes for version 3.2 (2009-06-05):
+----------------------------------------------------------------------
 
  ** Compatibility with Matlab graphics has been improved.
 
     The hggroup object and associated listener callback functions have
     been added allowing the inclusion of group objects.  Data sources
     have been added to these group objects such that
 
            x = 0:0.1:10;
@@ -1513,18 +1513,18 @@ Summary of important user-visible change
       lchol               spdiag     spsum
       loadimage           spfind     spsumsq
       mark_as_command     spinv      str2mat
       mark_as_rawcommand  spkron     unmark_command
       spatan2             splchol    unmark_rawcommand
       spchol              split
       spchol2inv          splu
 
-Summary of important user-visible changes for version 3.0:
----------------------------------------------------------
+Summary of important user-visible changes for version 3.0 (2007-12-21):
+----------------------------------------------------------------------
 
  ** Compatibility with Matlab graphics is much better now.  We now
     have some graphics features that work like Matlab's Handle
     Graphics (tm):
 
     + You can make a subplot and then use the print function to
       generate a file with the plot.
 
